components/resources/quizzes/shared/Progress.tsx

```diff
@@ -1,14 +1,16 @@
 import React from "react";
 type ProgressProps = {
   progress: number;
+  index: number;
 };
 
-const Progress: React.FC<ProgressProps> = ({ progress }) => {
+const Progress: React.FC<ProgressProps> = ({ progress, index }) => {
   return (
     <div className="relative w-full h-4 bg-gray-300 rounded-full">
       <div
         className="absolute top-0 left-0 h-full rounded-full bg-violet-400"
-        style={{ width: `${progress}%` }}
+        // custom progress bar styling, so that the progress bar automatically starts at zero and evenly increments to 100 at the BluePrint
+        style={{ width: `${index >= 0 ? (index * 100) / progress : 100}%` }}
```

Comment:
I think all this calculation should be happening in the view state, not in the component. `progress` should be a percentage that you just pass in. This component shouldnt need to know the current index. You're just passing that in here to do you calculation. If you do the calculation in the state, it's easiest to test as well.